### ISSUE TITLE
Implement bias reduction method

### ISSUE THREAD

**Issue by proteneer** [deepchem/deepchem#617]
https://arxiv.org/pdf/1706.06619.pdf

In particular the AVE minimization algorithm uses a genetic algorithm to minimize the bias (comprised of both active and inactive measures)

**Comment by lilleswing** [comment]
![selection_035](https://user-images.githubusercontent.com/1225605/27554030-1ea18e66-5a7b-11e7-85b5-7e1c18caa40d.png)

**Comment by rbharath** [comment]
The authors said they might possibly contribute an implementation to DeepChem. Would be cool to see how our benchmark suite behaves with AVE

**Comment by patrickhop** [comment]
would love to see this added

**Comment by rbharath** [comment]
@lilleswing Swapping this over to an intermediate level contribution since I think some work will be involved.

This would be a good project for those with interest in learning about proper statistical techniques for scientific applications.

**Comment by anushkagupta200615-jpg** [comment]
@lilleswing 

I have worked on implementing the bias reduction method discussed in this issue based on the paper Most Ligand-Based Benchmarks Measure Overfitting Rather than Accuracy (arxiv:1706.06619).

I've added a new AVESplitter to deepchem.splits which successfully utilizes the Genetic Algorithm to minimize the nearest-neighbor distance discrepancies between the active/inactive training and validation sets. I tested it locally on the Tox21 dataset with ECFP fingerprints, and it successfully drove the AVE bias down to near-zero (< 0.01).

I've also added unit tests to ensure there's no data leakage.

I have just opened a Pull Request with these changes. I would really appreciate it if one of the maintainers could go through my PR and review the implementation when you have the time! Thank you!
